operator-pipeline-images/operatorcert/entrypoints/detect_changed_operators.py

```diff
@@ -0,0 +1,245 @@
+            - extra_files: list of the names of the files that were affected
+                by the change but do not belong to an operator or a bundle
+            - affected_operators: list of the names of all operators affected
+                by the changes
+            - added_operators: list of the names of new operators that were
+                added by the changes
+            - modified_operators: list of the names of existing operators that
+                were modified by the changes
+            - deleted_operators: list of the names of operators that were
+                removed by the changes
+            - affected_bundles: list of the names of all bundles affected by
+                the changes
+            - added_bundles: list of the names of new bundles that were added
+                by the changes
+            - modified_bundles: list of the names of existing bundles that
+                were modified by the changes
+            - deleted_bundles: list of the names of bundles that were deleted
+                by the changes
+    """
+
+    def _find_owner(path: str) -> Tuple[Optional[str], Optional[str]]:
+        """
+        Given a relative file name within an operator repo, return a pair
+        (operator_name, bundle_version) indicating what bundle the file belongs to.
+        If the file is outside a bundle directory, the returned bundle_version will
+        be None.
+        If the file is outside an operator directory, both operator_name and
+        bundle_version will be None.
+        Args:
+            path: Path to the file
+        Returns:
+            (operator_name, bundle_version) Operator/bundle the files belongs to
+        """
+        # split the relative filename into a tuple of individual components
+        filename_parts = pathlib.Path(path).parts
+        if len(filename_parts) < 3 or filename_parts[0] != "operators":
+            # path is outside an operator directory
+            return None, None
+        _, detected_operator_name, detected_bundle_version, *rest = filename_parts
+        if len(rest) < 1:
+            # inside an operator but not in a bundle (i.e.: the ci.yaml file)
+            return detected_operator_name, None
+        return detected_operator_name, detected_bundle_version
+
+    LOGGER.debug("Detecting changes in repo at %s", repo_path)
+
+    with tempfile.TemporaryDirectory() as repo_clone_path:
+        # Phase 1: using the diff between PR head and base, detect which bundles
+        # have been affected by the PR
+
+        git_repo_head = GitRepo(repo_path)
+        # Ensure the head commit is checked out (mostly needed for tests)
+        git_repo_head.head.reference = git_repo_head.create_head(
+            "pr_head", head_commit, force=True
+        )
+        git_repo_head.head.reset(index=True, working_tree=True)
+        # Make a temporary clone of the repo and check out the base commit
+        git_repo_base = git_repo_head.clone(repo_clone_path)
```

Comment:
Do we need to clone the repo again if the previous pipeline task already cloned it?